dotnet/src/Connectors/Connectors.HuggingFace/Core/Models/ChatCompletionRequest.cs

```diff
@@ -118,7 +119,7 @@ internal static ChatCompletionRequest FromChatHistoryAndExecutionSettings(ChatHi
             Temperature = executionSettings.Temperature,
             Stop = executionSettings.Stop,
             MaxTokens = executionSettings.MaxTokens,
-            Model = executionSettings.ModelId ?? TextGenerationInferenceDefaultModel,
+            Model = executionSettings.ModelId ?? modelId ?? TextGenerationInferenceDefaultModel,
```

Comment:
If the model id is provided in the method I would assume it has priority.

```suggestion
            Model = modelId ?? executionSettings.ModelId ?? TextGenerationInferenceDefaultModel,
```

Reply:
Behaviour explained in the param doc: `<param name="modelId">Model id to use if value in prompt execution settings is not set.</param>`

1. `executionSettings.ModelId` is what was set when the service was invoked
2. `modelId` is what is configured for the service
3. `TextGenerationInferenceDefaultModel` is the fallback